shortestPathInBinaryMatrix.js

```diff
@@ -0,0 +1,111 @@
+/*
+Given an n x n binary matrix grid, return the length of the shortest clear path in the matrix. If there is no clear path, return -1.
+
+A clear path in a binary matrix is a path from the top-left cell (i.e., (0, 0)) to the bottom-right cell (i.e., (n - 1, n - 1)) such that:
+
+All the visited cells of the path are 0.
+All the adjacent cells of the path are 8-directionally connected (i.e., they are different and they share an edge or a corner).
+The length of a clear path is the number of visited cells of this path.
+*/
+
+const { Queue } = require('./utils/queue');
+const PATH = 0;
+const NO_CLEAR_PATH = -1;
+
+/*
+
+   0  1  2 
+0 [0, 0, 0],
+1 [1, 1, 0],
+2 [1, 1, 0],
+
+*/
+
+const directions = [
+    [-1, -1],
+    [-1, 0],
+    [-1, 1],
+    [0, -1],
+    [0, 1],
+    [1, -1],
+    [1, 0],
+    [1, 1],
+];
+
+const isInBoundWithRowAndCol = (grid, newRow, newCol) => {
```

Comment:
I don't see why you need the `withRowAndCol` part in the name. I think `isInBounds` is clear and also generic.